`Error: Unsupported language version '0.8.27'`

For repro use my repo here: https://github.com/pcaversaccio/hardhat-project-template-ts

I updated the Solidity version to `0.8.27` via https://github.com/pcaversaccio/hardhat-project-template-ts/commit/f44a0350d47e26a56666a175d2abf9559e7eac0b.

The hardhat tests currently fail due to your package not supporting the Solidity version `0.8.27`: `npx hardhat test`:

```console
An unexpected error occurred:
Error: Unsupported language version '0.8.27'.
    at tryRemoveNonStructNatSpec (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/@openzeppelin+upgrades-core@1.37.0/node_modules/@openzeppelin/upgrades-core/src/utils/make-namespaced.ts:[18](https://github.com/blokk-studio/hardhat-project-template-ts/actions/runs/10752994779/job/29821936631#step:10:19)3:22)
    at makeNamespacedInput (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/@openzeppelin+upgrades-core@1.37.0/node_modules/@openzeppelin/upgrades-core/src/utils/make-namespaced.ts:158:28)
    at OverriddenTaskDefinition._action (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/@openzeppelin+hardhat-upgrades@3.2.1_@nomicfoundation+hardhat-ethers@3.0.8_ethers@6.13.2_buff_b6fk7igjaxyvtwxdu22yysbakm/node_modules/@openzeppelin/hardhat-upgrades/src/index.ts:102:31)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Environment._runTaskDefinition (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/hardhat@2.22.10_bufferutil@4.0.8_ts-node@10.9.2_@types+node@22.5.4_typescript@5.5.4__typescri_7iajnuxsvogcbxkb6ol3j2urla/node_modules/hardhat/src/internal/core/runtime-environment.ts:351:14)
    at async Environment.run (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/hardhat@2.[22](https://github.com/blokk-studio/hardhat-project-template-ts/actions/runs/10752994779/job/29821936631#step:10:23).10_bufferutil@4.0.8_ts-node@10.9.2_@types+node@22.5.4_typescript@5.5.4__typescri_7iajnuxsvogcbxkb6ol3j2urla/node_modules/hardhat/src/internal/core/runtime-environment.ts:184:14)
    at async SimpleTaskDefinition.action (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/hardhat@2.22.10_bufferutil@4.0.8_ts-node@10.9.2_@types+node@22.5.4_typescript@5.5.4__typescri_7iajnuxsvogcbxkb6ol3j2urla/node_modules/hardhat/src/builtin-tasks/compile.ts:1024:37)
    at async Environment._runTaskDefinition (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/hardhat@2.22.10_bufferutil@4.0.8_ts-node@10.9.2_@types+node@22.5.4_typescript@5.5.4__typescri_7iajnuxsvogcbxkb6ol3j2urla/node_modules/hardhat/src/internal/core/runtime-environment.ts:351:14)
    at async Environment.run (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/hardhat@2.22.10_bufferutil@4.0.8_ts-node@10.9.2_@types+node@22.5.4_typescript@5.5.4__typescri_7iajnuxsvogcbxkb6ol3j2urla/node_modules/hardhat/src/internal/core/runtime-environment.ts:184:14)
    at async /home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/hardhat@2.22.10_bufferutil@4.0.8_ts-node@10.9.2_@types+node@22.5.4_typescript@5.5.4__typescri_7iajnuxsvogcbxkb6ol3j2urla/node_modules/hardhat/src/builtin-tasks/compile.ts:464:28 {
  code: 'GenericFailure'
}
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

I don't use your package at all, but it's required by the `@tenderly/hardhat-tenderly` package unfortunately. I haven't debugged what is triggering exactly this issue, but would appreciate a patch :)

--- a/packages/core/CHANGELOG.md
+++ b/packages/core/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Changelog
 
+## Unreleased
+
+- Fix Hardhat compile error when using solc version `0.8.27`. ([#1075](https://github.com/OpenZeppelin/openzeppelin-upgrades/pull/1075))
+
 ## 1.37.0 (2024-08-28)
 
 - **Breaking change**: CLI: Disallow self-references for storage layout validations. ([#1067](https://github.com/OpenZeppelin/openzeppelin-upgrades/pull/1067))

--- a/packages/core/CHANGELOG.md
+++ b/packages/core/CHANGELOG.md
@@ -1,6 +1,6 @@
 # Changelog
 
-## Unreleased
+## 1.37.1 (2024-09-09)
 
 - Fix Hardhat compile error when using solc version `0.8.27`. ([#1075](https://github.com/OpenZeppelin/openzeppelin-upgrades/pull/1075))
 

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@openzeppelin/upgrades-core",
-  "version": "1.37.0",
+  "version": "1.37.1",
   "description": "",
   "repository": "https://github.com/OpenZeppelin/openzeppelin-upgrades/tree/master/packages/core",
   "license": "MIT",

--- a/packages/core/src/utils/make-namespaced.ts
+++ b/packages/core/src/utils/make-namespaced.ts
@@ -164,7 +164,7 @@ export function makeNamespacedInput(input: SolcInput, output: SolcOutput, solcVe
 }
 
 /**
- * If Slang is supported for the current platform and we have the compiler version available,
+ * If we have the compiler version available and Slang is supported for the current platform and compiler version,
  * use Slang to parse and remove all NatSpec comments that do not precede a struct definition and return the modified content.
  *
  * Otherwise, return the original content.
@@ -172,7 +172,7 @@ export function makeNamespacedInput(input: SolcInput, output: SolcOutput, solcVe
 function tryRemoveNonStructNatSpec(origContent: string, solcVersion: string | undefined): string {
   const natSpecRemovals: Modification[] = [];
 
-  if (solcVersion !== undefined && tryRequire('@nomicfoundation/slang')) {
+  if (solcVersion !== undefined && slangSupportsPlatformAndVersion(solcVersion)) {
     /* eslint-disable @typescript-eslint/no-var-requires */
     const { Language } = require('@nomicfoundation/slang/language');
     const { NonterminalKind, TerminalKind } = require('@nomicfoundation/slang/kinds');
@@ -215,6 +215,21 @@ function tryRequire(id: string) {
     // do nothing
   }
   return false;
+}
+
+/**
+ * Whether Slang is supported for the current platform and the given compiler version.
+ */
+function slangSupportsPlatformAndVersion(solcVersion: string): boolean {
+  if (tryRequire('@nomicfoundation/slang')) {
+    /* eslint-disable @typescript-eslint/no-var-requires */
+    const { Language } = require('@nomicfoundation/slang/language');
+    /* eslint-enable @typescript-eslint/no-var-requires */
+
+    return Language.supportedVersions().includes(solcVersion);
+  } else {
+    return false;
+  }
 }
 
 interface Modification {

--- a/packages/core/src/utils/make-namespaced.ts
+++ b/packages/core/src/utils/make-namespaced.ts
@@ -172,7 +172,7 @@ export function makeNamespacedInput(input: SolcInput, output: SolcOutput, solcVe
 function tryRemoveNonStructNatSpec(origContent: string, solcVersion: string | undefined): string {
   const natSpecRemovals: Modification[] = [];
 
-  if (solcVersion !== undefined && slangSupportsPlatformAndVersion(solcVersion)) {
+  if (solcVersion !== undefined && tryRequire('@nomicfoundation/slang') && slangSupportsVersion(solcVersion)) {
     /* eslint-disable @typescript-eslint/no-var-requires */
     const { Language } = require('@nomicfoundation/slang/language');
     const { NonterminalKind, TerminalKind } = require('@nomicfoundation/slang/kinds');
@@ -217,19 +217,12 @@ function tryRequire(id: string) {
   return false;
 }
 
-/**
- * Whether Slang is supported for the current platform and the given compiler version.
- */
-function slangSupportsPlatformAndVersion(solcVersion: string): boolean {
-  if (tryRequire('@nomicfoundation/slang')) {
-    /* eslint-disable @typescript-eslint/no-var-requires */
-    const { Language } = require('@nomicfoundation/slang/language');
-    /* eslint-enable @typescript-eslint/no-var-requires */
+function slangSupportsVersion(solcVersion: string): boolean {
+  /* eslint-disable @typescript-eslint/no-var-requires */
+  const { Language } = require('@nomicfoundation/slang/language');
+  /* eslint-enable @typescript-eslint/no-var-requires */
 
-    return Language.supportedVersions().includes(solcVersion);
-  } else {
-    return false;
-  }
+  return Language.supportedVersions().includes(solcVersion);
 }
 
 interface Modification {

